src/main/js/main_html5.js

```diff
@@ -1488,24 +1609,63 @@ require("../../../html5-common/js/utils/environment.js");
       }
 
       var buffer = 0;
-      if (event.target.buffered && event.target.buffered.length > 0) {
-        buffer = event.target.buffered.end(0); // in sec;
-      }
-
-      // durationchange event raises the currentTime as a string
-      var resolvedTime = (event && event.target) ? event.target.currentTime : null;
-      if (resolvedTime && (typeof resolvedTime !== "number")) {
-        resolvedTime = Number(resolvedTime);
+      var newCurrentTime = null;
+      var currentLiveTime = 0;
+      var duration = resolveDuration(event.target.duration);
+
+      // Live videos without DVR (i.e. maxTimeShift === 0) are treated as regular
+      // videos for playhead update purposes
+      if (isDvrAvailable()) {
+        var maxTimeShift = getMaxTimeShift();
+        newCurrentTime = currentTimeShift - maxTimeShift;
+        duration = -maxTimeShift;
+        buffer = duration;
+        // [PBW-5863] The skin displays current time a bit differently when dealing
+        // with live video, but we still need to keep track of the actual playhead for analytics purposes
+        currentLiveTime = _video.currentTime;
+      } else {
+        if (_video.buffered && _video.buffered.length > 0) {
+          buffer = _video.buffered.end(0); // in seconds
+        }
+        // Just a precaution for older browsers, this should already be a number
+        newCurrentTime = ensureNumber(_video.currentTime, null);
       }
 
       var seekable = getSafeSeekRange(getSafeSeekableObject());
-      this.controller.notify(eventname,
-                             { "currentTime": resolvedTime,
-                               "duration": resolveDuration(event.target.duration),
-                               "buffer": buffer,
-                               "seekRange": seekable });
+      this.controller.notify(eventname, {
+        currentTime: newCurrentTime,
+        currentLiveTime: currentLiveTime,
+        duration: duration,
+        buffer: buffer,
+        seekRange: seekable
+      });
     }, this);
 
+    /**
+     * Converts a value to a number or returns null if it can't be converted or is not a finite value.
+     * @private
+     * @method OoyalaVideoWrapper#ensureNumber
+     * @param {*} value The value to convert.
+     * @param {*} defaultValue A default value to return when the input is not a valid number.
+     * @return {Number} The Number equivalent of value if it can be converted and is finite.
+     * When value doesn't meet the criteria the function will return either defaultValue (if provided) or null.
+     */
+    var ensureNumber = function(value, defaultValue) {
```

Comment:
Doesn't have to be done now, but should we move this to html5-common since this is basically a copy of the function that's in html5-skin?

Reply:
I was sort of hoping no one would notice lol. I think we can probaly move this after we work on the html5-common tech debt task.